swo_aws_extension/flows/jobs/billing_journal/billing_journal_generator.py

```diff
@@ -764,3 +821,80 @@ def _generate_zip(self):
                     zip_file.writestr(filename, file_content)
         zip_buffer.seek(0)
         return zip_buffer
+
+    def _manage_invalid_services_by_account(
+        self, account_metrics, journal_details, account_invoices, account_id
+    ):
+        """Checks for services in the account that do not match any subscription item.
+
+        Args:
+            account_metrics (dict): Metrics for the account.
+            journal_details (dict): Journal metadata.
+            account_invoices (dict): Invoices for the account.
+            account_id (str): AWS account ID.
+        """
+        account_reports = self.organization_reports["accounts"].get(account_id, {})
+        if not account_reports:
+            return
+        services_by_account = self._get_services_amounts_by_account(account_reports)
+        partner_discount = account_metrics.get(UsageMetricTypeEnum.PROVIDER_DISCOUNT.value, {})
+
+        for service, amount in services_by_account.items():
+            service_exist = False
+            for line in self.journal_file_lines:
+                service_name = line.description.value1
+                service_account_id = line.description.value2.split("/")[0]
+                mpa_account_id = line.externalIds.vendor
+                if service_name == service and account_id in [service_account_id, mpa_account_id]:
+                    service_exist = True
+                    break
+
+            if not service_exist:
+                service_discount = partner_discount.get(service, 0)
+                partner_amount = amount - abs(service_discount)
+                discount = ((amount - partner_amount) / amount) * 100 if amount != 0 else 0
+                error_msg = (
+                    f"{account_id} - Service {service} with amount {amount} and discount "
+                    f"{discount} did not match any subscription item."
+                )
+                self._log("error", error_msg)
+                error_line = self._create_line_error(
+                    account_id,
+                    "Error",
+                    account_metrics,
+                    journal_details,
+                    account_invoices,
+                    service,
+                    amount,
+                    error_msg,
+                )
+                self.journal_file_lines.append(error_line)
+
+    @staticmethod
+    def _get_services_amounts_by_account(account_reports):
+        """
+        Extracts services and their amounts from the account reports.
+        Args:
+            account_reports (dict): Account reports containing service data.
+
+        Returns:
+            dict: A dictionary mapping service names to their amounts.
+
+        """
+        services_by_account = {}
+        record_type = account_reports.get(
+            JournalAttachmentFilesNameEnum.RECORD_TYPE_AND_SERVICE_COST
```

Comment:
A suggestion to use `.value` here. I personally started using it everywhere just to be on safe side